silkworm/rpc/http/connection.cpp

```diff
@@ -62,213 +58,172 @@ Connection::~Connection() {
 
 Task<void> Connection::read_loop() {
     try {
-        // Read next request or next chunk (result == RequestParser::indeterminate) until closed or error
         while (true) {
             co_await do_read();
         }
     } catch (const boost::system::system_error& se) {
-        if (se.code() == boost::asio::error::eof || se.code() == boost::asio::error::connection_reset || se.code() == boost::asio::error::broken_pipe) {
+        if (se.code() == boost::beast::http::error::end_of_stream || se.code() == boost::asio::error::broken_pipe) {
             SILK_DEBUG << "Connection::read_loop close from client with code: " << se.code();
         } else if (se.code() != boost::asio::error::operation_aborted) {
             SILK_ERROR << "Connection::read_loop system_error: " << se.what();
-            std::rethrow_exception(std::make_exception_ptr(se));
+            throw;
         } else {
             SILK_DEBUG << "Connection::read_loop operation_aborted: " << se.what();
         }
     } catch (const std::exception& e) {
         SILK_ERROR << "Connection::read_loop exception: " << e.what();
-        std::rethrow_exception(std::make_exception_ptr(e));
+        throw;
     }
 }
 
 Task<void> Connection::do_read() {
     SILK_DEBUG << "Connection::do_read going to read...";
-    std::size_t bytes_read = co_await socket_.async_read_some(boost::asio::buffer(buffer_), boost::asio::use_awaitable);
-    SILK_DEBUG << "Connection::do_read bytes_read: " << bytes_read;
-    SILK_TRACE << "Connection::do_read buffer: " << std::string_view{static_cast<const char*>(buffer_.data()), bytes_read};
-
-    RequestParser::ResultType result = request_parser_.parse(request_, buffer_.data(), buffer_.data() + bytes_read);
-
-    if (result == RequestParser::ResultType::good) {
-        co_await handle_request(request_);
-        clean();
-    } else if (result == RequestParser::ResultType::bad) {
-        reply_ = Reply::stock_reply(StatusType::bad_request);
-        co_await do_write();
-        clean();
-    } else if (result == RequestParser::ResultType::processing_continue) {
-        reply_ = Reply::stock_reply(StatusType::processing_continue);
-        co_await do_write();
-        reply_.reset();
+
+    boost::beast::http::request_parser<boost::beast::http::string_body> parser;
+    unsigned long bytes_transferred;
+    try {
+        bytes_transferred = co_await boost::beast::http::async_read(socket_, data_, parser, boost::asio::use_awaitable);
+    } catch (const boost::system::system_error& se) {
+        if (se.code() == boost::beast::http::error::end_of_stream || se.code() == boost::asio::error::broken_pipe) {
+            throw;
+        } else {
+            co_return;
+        }
+    }
+
+    SILK_DEBUG << "Connection::do_read bytes_read: " << bytes_transferred;
```

Comment:
`SILK_DEBUG` -> `SILK_TRACE` in this log line

Reply:
ok